internal/binarylog/method_logger.go

```diff
@@ -52,6 +53,23 @@ type MethodLogger interface {
 	Log(LogEntryConfig)
 }
 
+// MethodLoggerWithContext is an optional interface that a method logger can
+// implement that takes a context on Log calls.
+type MethodLoggerWithContext interface {
+	LogWithContext(context.Context, LogEntryConfig)
+}
+
+// BinLogWithContext is a helper to pass a context to the binary logger's Log
+// function if binary logger has support. If not, it calls the log method that
+// doesn't take a context.
+func BinLogWithContext(ctx context.Context, binLogger MethodLogger, lec LogEntryConfig) {
```

Comment:
Nit: `LogWithContext`?  `binarylog.BinLog` doesn't strictly stutter in Go terms, but it realistically does.

Reply:
Fair point. However, with the refactor of this we discussed offline (won't have this anymore, everything gets plumbed with logging calls that take context, SetLogger keep backwards compatible and simply wraps it.

Reply:
Actually, implemented solution below, but this function still got deleted, so this comment no longer applies.

Reply:
Actually, discussed offline in QAZSE, switched the branch to SetLogger() called at init() time.